libs/ui-watt/src/lib/components/input/timepicker/watt-timepicker.component.ts

```diff
@@ -121,23 +226,45 @@ export class WattTimepickerComponent extends WattPickerBase {
    */
   protected initRangeInput() {
     // Setup and subscribe for input changes
-    this.rangeInputService.init({
-      startInput: this.maskInput(
-        this.startInput.nativeElement,
-        (this.initialValue as WattRange | null)?.start
-      ),
-      endInput: this.maskInput(
-        this.endInput.nativeElement,
-        (this.initialValue as WattRange | null)?.end
-      ),
+    const startInput = this.maskInput(
+      this.startInput.nativeElement,
+      (this.initialValue as WattRange | null)?.start
+    );
+
+    const endInput = this.maskInput(
+      this.endInput.nativeElement,
+      (this.initialValue as WattRange | null)?.end
+    );
+
+    this.rangeInputService.init({ startInput, endInput });
+
+    // Silence the compiler since `onInputChanges$` is always assigned in `init`
+    const { onInputChanges$ = EMPTY } = this.rangeInputService;
+    const timeRange$ = onInputChanges$.pipe(takeUntil(this.destroy$));
+
+    // Synchronize the slider value with the input fields. Calling `update`
+    // here automatically triggers an emit on the `timeRange$` observable.
+    this.sliderChange$.pipe(takeUntil(this.destroy$)).subscribe((value) => {
+      startInput.maskedInput.update(minutesToTime(value.min));
+      endInput.maskedInput.update(minutesToTime(value.max));
     });
 
-    this.rangeInputService.onInputChanges$
-      ?.pipe(takeUntil(this.destroy$))
-      .subscribe(([start, end]) => {
-        this.markParentControlAsTouched();
-        this.changeParentValue({ start, end });
-      });
+    // Whenever the start input value changes, set (or remove)
+    // the minimum allowed value for the end input value.
+    timeRange$
+      .pipe(map(getTruthy(0)), distinctUntilChanged())
```

Comment:
suggestion: wouldn't this give the same result?

```suggestion
      .pipe(map(([start]) => start || undefined), distinctUntilChanged())
```

Reply:
Yes, but hurts readability IMO. The helper was just meant to "DRY" it a little while also explicitly expressing that the truthy check is important here, otherwise one might be inclined to remove the `|| undefined` part when visiting the component at a later time. But I'm also an FP nerd and what is readable to me might of course not be that readable to others 🙂

Reply:
The helper function looks fine. One thing that stands out is the index param. It feels a bit like a magic number. What do you think about renaming the function to `getTruthyByIndex` (or similar) so the `0` and `1` have some meaning?

Reply:
Good point! I have renamed it to `getTruthyAt`, that way it is similar to [`Array.prototype.at`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/at).

Reply:
Even better 🙂